manifests/certificate/x509.pp

```diff
@@ -118,6 +118,7 @@
   Boolean                        $force = true,
   String                         $cnf_tpl = 'openssl/cert.cnf.erb',
   Boolean                        $encrypted = true,
+  Boolean                        $use_req,
```

Comment:
Actually, the idea was that `$use_req` would be a dynamic variable, not a parameter.

Typically, it would be `$use_req = !empty($altnames+$extkeyusage)` since that seems to define whether to use `req_extensions`.

Reply:
OK. So I guess I didn't really see the need to have that on top of having `req_ext` already. Since it is my first attempt, I assumed (wrongly) that the scope of `req_ext` was not suitable. I re-submitted taking out completely `use_req` and use the existing `req_ext` as to me the semantic of the variable is pretty much the same.

Apologies for the whole mess for this little thing. I am however making sure to remember all that stuff for later. Thanks for your guidance.